### PR TITLE
genpolicy: reject untested CreateContainer field values

### PR DESCRIPTION
**Pull request by danmihai1**
Reject CreateContainerRequest field values that are not tested by Kata CI and that might impact the confidentiality of CoCo Guests.

This change uses a "better safe than sorry" approach to untested fields. It is very possible that in the future we'll encounter
reasonable use cases that will either:

- Show that some of these fields are benign and don't have to be verified by Policy, or
- Show that Policy should verify legitimate values of these fields

These are the new CreateContainerRequest Policy rules:

    count(input.shared_mounts) == 0
    is_null(input.string_user)

    i_oci := input.OCI
    is_null(i_oci.Hooks)
    is_null(i_oci.Linux.Seccomp)
    is_null(i_oci.Solaris)
    is_null(i_oci.Windows)

    i_linux := i_oci.Linux
    count(i_linux.GIDMappings) == 0
    count(i_linux.MountLabel) == 0
    count(i_linux.Resources.Devices) == 0
    count(i_linux.RootfsPropagation) == 0
    count(i_linux.UIDMappings) == 0
    is_null(i_linux.IntelRdt)
    is_null(i_linux.Resources.BlockIO)
    is_null(i_linux.Resources.Network)
    is_null(i_linux.Resources.Pids)
    is_null(i_linux.Seccomp)
    i_linux.Sysctl == {}

    i_process := i_oci.Process
    count(i_process.SelinuxLabel) == 0
    count(i_process.User.Username) == 0

### CODE REVIEW

File: src/tools/genpolicy/rules.rego [hunk 1]
```diff
@@ -85,6 +90,38 @@ CreateContainerRequest {
...
+    i_oci := input.OCI
+    is_null(i_oci.Hooks)
+    is_null(i_oci.Linux.Seccomp)
+    is_null(i_oci.Solaris)
+    is_null(i_oci.Windows)
+
+    i_linux := i_oci.Linux
+    count(i_linux.GIDMappings) == 0
+    count(i_linux.MountLabel) == 0
+    count(i_linux.Resources.Devices) == 0
```
Comment (burgerdev) [inline src/tools/genpolicy/rules.rego:108]: As mentioned in https://github.com/kata-containers/kata-containers/pull/9706#discussion_r1624683480, I think this is going to break device plugins. I'm not too concerned about that yet, but that may change when people want to use GPUs or something :)

 I have some code with which I can verify that on Monday.

**Review by burgerdev — COMMENTED** [review]
Thanks, this is a great addition!

File: src/tools/genpolicy/rules.rego (hunk 1 above)
Comment (danmihai1) [reply src/tools/genpolicy/rules.rego:108]: Thanks! Sure, if we have broken use cases, let's fix them.

Comment (danmihai1) [reply src/tools/genpolicy/rules.rego:107]: @burgerdev, do you think we should merge this PR? Or should we wait for additional experiments?

Thanks!

Comment (burgerdev) [reply src/tools/genpolicy/rules.rego:107]: Sorry for the delay. Things I verified to keep working:

* Adding a character device via device plugin
* Adding a block device via CSI

None of these sets `i_linux.Resources.Devices`, no idea how I arrived at that expectation. Anyway, please go ahead with the merge!

Comment (danmihai1) [reply src/tools/genpolicy/rules.rego:107]: No worries, thanks for helping out!

**Review by sprt — APPROVED** [review]

**Review by GabyCT — APPROVED** [review]

**Pull request merged**